### PR TITLE
Initial commit of VIS based Vehicle HAL

### PR DESCRIPTION
**Pull request by arminn**
Signed-off-by: Andrii Chepurnyi <andrii_chepurnyi@epam.com>

### CODE REVIEW

**Review by rshym — APPROVED** [review]

File: common/src/VisClient.cpp [hunk 1]
```diff
@@ -0,0 +1,825 @@
...
+    return 0;
+}
+
+int VisClient::init() {
+    std::lock_guard<std::mutex> lock(mLock);
+    ALOGV("init");
+
+    if (mUri.empty()) {
+        char propValue[PROPERTY_VALUE_MAX] = {};
+        property_get("persist.vis.uri", propValue, "wss://192.168.1.100:8088");
```
Comment (rshym) [inline common/src/VisClient.cpp]: Is it expected to be hard coded like this?
Maybe put it as define to top of file with explanation in comments?

File: common/src/VisClient.cpp [hunk 2]
```diff
@@ -0,0 +1,825 @@
...
+    mConnectedState = ConnState::STATE_DISCONNECTED;
+    mWs = nullptr;
+}
+
+void VisClient::handleError(void* user) {
+    ALOGV("handleError");
+    std::lock_guard<std::mutex> lock(mLock);
+
+    switch (reinterpret_cast<size_t>(user)) {
+        case 1:
```
Comment (rshym) [inline common/src/VisClient.cpp]: As I understand these error codes are defined somewhere outside of project.
If this is the case, we may provide comment about source of these numbers, for possible expansion and verification.

File: common/src/VisVehicleHal.cpp [hunk 3]
```diff
@@ -0,0 +1,629 @@
...
+                              it->second.area, it->first.c_str());
+                    }
+                }
+            } else {
+                ALOGE("Unable to read current value for prop 0x%x area=0x%x from propertystore",
+                      it->second.prop, it->second.area);
+            }
+        }
+    }
+#if 0
```
Comment (rshym) [inline common/src/VisVehicleHal.cpp:235]: Do we need to keep disabled code?
If it is intended to be used in some cases, may be to add some comments regarding possible use case.

Comment (arminn) [reply common/src/VisVehicleHal.cpp:235]: In short - yes. We need it for debug/simulation.

**Review by rshym — APPROVED** [review]

**Pull request merged**